internal/services/workloads/workloads_sap_virtual_instance_resource.go

```diff
@@ -0,0 +1,1129 @@
+									ForceNew: true,
+									MaxItems: 1,
+									Elem: &pluginsdk.Resource{
+										Schema: map[string]*pluginsdk.Schema{
+											"application_server": {
+												Type:     pluginsdk.TypeList,
+												Optional: true,
+												ForceNew: true,
+												MaxItems: 1,
+												Elem: &pluginsdk.Resource{
+													Schema: map[string]*pluginsdk.Schema{
+														"availability_set_name": {
+															Type:         pluginsdk.TypeString,
+															Optional:     true,
+															ForceNew:     true,
+															ValidateFunc: validate.AvailabilitySetName,
+														},
+
+														"virtual_machine": SchemaForSAPVirtualInstanceVirtualMachineFullResourceNames(),
+													},
+												},
+											},
+
+											"central_server": {
+												Type:     pluginsdk.TypeList,
+												Optional: true,
+												ForceNew: true,
+												MaxItems: 1,
+												Elem: &pluginsdk.Resource{
+													Schema: map[string]*pluginsdk.Schema{
+														"availability_set_name": {
+															Type:         pluginsdk.TypeString,
+															Optional:     true,
+															ForceNew:     true,
+															ValidateFunc: validate.AvailabilitySetName,
+														},
+
+														"load_balancer": SchemaForSAPVirtualInstanceLoadBalancerFullResourceNames(),
+
+														"virtual_machine": SchemaForSAPVirtualInstanceVirtualMachineFullResourceNames(),
+													},
+												},
+											},
+
+											"database_server": {
+												Type:     pluginsdk.TypeList,
+												Optional: true,
+												ForceNew: true,
+												MaxItems: 1,
+												Elem: &pluginsdk.Resource{
+													Schema: map[string]*pluginsdk.Schema{
+														"availability_set_name": {
+															Type:         pluginsdk.TypeString,
+															Optional:     true,
+															ForceNew:     true,
+															ValidateFunc: validate.AvailabilitySetName,
+														},
+
+														"load_balancer": SchemaForSAPVirtualInstanceLoadBalancerFullResourceNames(),
+
+														"virtual_machine": SchemaForSAPVirtualInstanceVirtualMachineFullResourceNames(),
+													},
+												},
+											},
+
+											"shared_storage": {
+												Type:     pluginsdk.TypeList,
+												Optional: true,
+												ForceNew: true,
+												MaxItems: 1,
+												Elem: &pluginsdk.Resource{
+													Schema: map[string]*pluginsdk.Schema{
+														"account_name": {
+															Type:         pluginsdk.TypeString,
+															Optional:     true,
+															ForceNew:     true,
+															ValidateFunc: storageValidate.StorageAccountName,
+														},
+
+														"private_endpoint_name": {
+															Type:         pluginsdk.TypeString,
+															Optional:     true,
+															ForceNew:     true,
+															ValidateFunc: networkValidate.PrivateLinkName,
+														},
```

Comment:
Can these be IDs as well

Reply:
It means service API will create new storage account and private endpoint per name when name is specified. So I assume we can't convert it to xxx_id.

Reply:
Since the API is creating resources on behalf of the user, can you verify that it also cleans them up/deletes them reliably when the sap virtual instance is deleted?

Reply:
Service API wouldn't delete them when SAP Virtual Instance is deleted. Because service team and major customers who are requesting this new resource confirmed that it's expected behavior and they expect to keep them since they're the part of the SAP system so that service API wouldn't actively delete them. But we can set "app_resource_group_name" and then SAP system will be deployed to this resource group. Then we could use "prevent_deletion_if_contains_resources = false" to delete the resources which created by service API when tf destroy.

Note: I already added note in [resource doc](https://github.com/hashicorp/terraform-provider-azurerm/pull/22869/files#diff-70507e2107861912a3d52f1986cbedbd0e7ea6c5ad2fc91af1af7ed049956d51R225) for this thing.